h/static/scripts/directive/annotation.coffee

```diff
@@ -202,6 +208,10 @@ AnnotationController = [
       switch @action
         when 'create'
           updateDomainModel(model, @annotation)
+
+          if group.focusedGroup().hashid != '__public__'
```

Comment:
> This is confusing. How could the hashid be a readable string like __public__. Should the client 
> even know these are hashids? Shouldn't they just be opaque identifiers here, and the hash bit 
> is an implementation detail hidden within the server?

The client-side code needs some kind of unique way to identify the group (including the pseudo-group _Public_) that the annotation belongs to. For example when the user clicks on a group in the groups dropdown, the controller needs to communicate which group was clicked on to the service. Or when we display various bit of the annotations and their editing controls, different directives need to know what to show in certain parts of the UI. We can't use group names because they're not unique.

Hashids are the obvious choice and then we just use the fake hashid `"__public__"` for when there's no group selected / the public group is selected.

I _could_ try and find some other way to represent this. Like for example when _Public_ is selected, maybe `group.focusedGroup()` just returns `null`. But there are various different places in the code where we need to display the name _Public_ for this group and I'd want to encode that name in one central place.